ikalog/inputs/input.py

```diff
@@ -178,22 +172,23 @@ def _skip_frame_recorded(self):
     #
     # @return Image if capture succeeded. Otherwise None.
     def read_frame(self):
-        self.lock.acquire()
-        if not self.is_active():
-            self.lock.release()
-            return None
-
-        next_tick = None
-
         try:
-            if self.cap_recorded_video:
-                self._skip_frame_recorded()
-            else:
-                next_tick = self._skip_frame_realtime()
-
-            self._next_frame_func()
+            self.lock.acquire()
+            if not self.is_active():
+                return None
```

Comment:
It should be no problem, because self.lock.release() in the finally block (line:192-193) is called before return.